test/core/metaflow_test/cli_check.py

```diff
@@ -148,44 +147,46 @@ def list_cards(self, step, task, card_type=None):
         no_card_found_message = CardNotPresentException.headline
         try:
             card_data = self._list_cards(step, task=task, card_type=card_type)
-            card_data = json.loads(card_data)
         except subprocess.CalledProcessError as e:
-            if no_card_found_message in e.output.decode("utf-8").strip():
+            if no_card_found_message in e.stderr.decode("utf-8").strip():
                 card_data = None
             else:
                 raise e
         return card_data
 
     def _list_cards(self, step, task=None, card_type=None):
-        pathspec = "%s/%s" % (self.run_id, step)
-        if task is not None:
-            pathspec = "%s/%s/%s" % (self.run_id, step, task)
-        cmd = ["--quiet", "card", "list", pathspec, "--as-json"]
-        if card_type is not None:
-            cmd.extend(["--type", card_type])
-
-        return self.run_cli(cmd, capture_output=True, pipe_error_to_output=True).decode(
-            "utf-8"
-        )
+        with NamedTemporaryFile(dir=".") as f:
+            pathspec = "%s/%s" % (self.run_id, step)
+            if task is not None:
+                pathspec = "%s/%s/%s" % (self.run_id, step, task)
+            cmd = ["--quiet", "card", "list", pathspec, "--as-json", "--file", f.name]
+            if card_type is not None:
+                cmd.extend(["--type", card_type])
+
+            self.run_cli(cmd, capture_output=True)
+            with open(f.name, "r") as jsf:
+                return json.load(jsf)
 
     def get_card(self, step, task, card_type, card_hash=None, card_id=None):
-        cmd = [
-            "--quiet",
-            "card",
-            "get",
-            "%s/%s/%s" % (self.run_id, step, task),
-            "--type",
-            card_type,
-        ]
-
-        if card_hash is not None:
-            cmd.extend(["--hash", card_hash])
-        if card_id is not None:
-            cmd.extend(["--id", card_id])
-
-        return self.run_cli(cmd, capture_output=True, pipe_error_to_output=True).decode(
-            "utf-8"
-        )
+        with NamedTemporaryFile(dir=".") as f:
+            cmd = [
+                "--quiet",
+                "card",
+                "get",
+                "%s/%s/%s" % (self.run_id, step, task),
+                f.name,
+                "--type",
+                card_type,
+            ]
+
+            if card_hash is not None:
+                cmd.extend(["--hash", card_hash])
+            if card_id is not None:
+                cmd.extend(["--id", card_id])
+
+            self.run_cli(cmd, capture_output=True)
+            with open(f.name, "r") as jsf:
```

Comment:
I think we may need to specify encoding here as well (read and write maybe -- some older version of python seem to prefer this)